src/main/java/duke/helper/Parser.java

```diff
@@ -0,0 +1,213 @@
+                command.equals("find"))) {
+            throw new DukeInvalidCommandException();
+        }
+        if (command.equals("list")) {
+            StringBuilder message = new StringBuilder("Here are the tasks in your list:");
+            for (int i = 1; i <= taskList.size(); i++) {
+                int index = i - 1;
+                message.append("\n" + i + ". " + taskList.get(index).toString());
+            }
+            return message.toString();
+        }
+        if (command.equals("mark")) {
+            int index = Integer.parseInt(strArr[1]) - 1;
+            Task temp;
+            try{
+                temp = taskList.get(index);
+            } catch(IndexOutOfBoundsException e) {
+                throw new DukeInvalidIndexException();
+            }
+            temp.markDone();
+            return "Nice! I've marked this task as done:\n" + temp.toString();
+        }
+
+        if (command.equals("todo")) {
+            String title;
+            try{
+                title = input.substring("todo".length() + 1).trim();
+            }
+            catch (StringIndexOutOfBoundsException e){
+                throw new DukeEmptyArgumentException();
+            }
+            Task task = new Todo(title);
+            taskList.insert(task);
+            return "Got it. I've added this task:" + "\n" +
+            "   " + task.toString() + "\n" + "Now you have " + taskList.size() + " tasks in the list.";
+        }
+
+        if (command.equals("deadline")) {
+            if (strArr.length == 1) throw new DukeEmptyArgumentException();
+            if (hasKeyword(strArr, "/by")) {
+                String title;
+                String[] splitArr = input.split("/by", 2);
+                try{
+                    title = splitArr[0].substring("deadline".length() + 1).trim();
+                }
+                catch (StringIndexOutOfBoundsException e){
+                    throw new DukeEmptyArgumentException();
+                }
+                String time = splitArr[1].trim();
+                Task task;
+                if(isDate(time)) {
+                    LocalDate ld = LocalDate.parse(time);
+                    task = new Deadline(title, ld);
+                } else {
+                    task = new Deadline(title, time);
+                }
+                taskList.insert(task);
+                return "Got it. I've added this task:" + "\n" + "   " + task.toString() + "\n" + "Now you have " + taskList.size() + " tasks in the list.";
+            }
+            else {
+                throw new DukeMissingArgumentException("/by");
+            }
+        }
+        if (command.equals("event")) {
+            if (strArr.length == 1) throw new DukeEmptyArgumentException();
+            if (hasKeyword(strArr, "/at")) {
+                String title;
+                String[] splitArr = input.split("/at", 2);
+                try {
```

Comment:
Maybe can put the try to try all of ur code and catch the exception so dont have so many trys and catch (Just a suggestion, not sure if it will be more complicated to do this)
try {
// all your different commands
} catch (error1) {
} catch (error2) {...}